### ISSUE TITLE
Handle conflicting aliases explicitly

### ISSUE THREAD

**Issue by mmillsAtREPAY** [dotnet/command-line-api#1323]
When an option is given an alias that is a substring of another alias, the remainder of the longer alias is treated as an argument to the shorter alias's option.

In my case, I had defined `-e <bool>` as an alias for a `--encrypt <bool>` parameter, and `-ekf <key file>` as an alias for a `--encryption-key-file <key file>` parameter.

At runtime, different scenarios could play out, depending on the command line I'm running.

```
> MyApp.exe SomeCommand -e true -ekf .\public.pgp
Option '-e' expects a single argument but 2 were provided.
Option '-e' expects a single argument but 2 were provided.
Unrecognized command or argument '.\public.pgp'
```

```
> MyApp.exe SomeCommand -ekf .\public.pgp
'kf' was not matched. Did you mean '-ekf', or '-r', or '-e', or '-o', or '-h'?
Unrecognized command or argument 'kf'
Unrecognized command or argument '.\public.pgp'
```

I'm not sure what the right answer is, but the three options I can think of are
* `throw` during initialization when an alias starts with another alias
* update the parser to make sure that unbundling doesn't take place if a matching multi-character alias exists, or
* explicitly disallow aliases that match `^-[\w][\w]+$` and `throw` during initialization if one is defined

**Comment by jonsequitur** [comment]
Can you provide a code sample to reproduce the issue? The type and arity of the options might affect the outcome.

**Comment by mmillsAtREPAY** [comment]
I cloned the repo locally and was trying to put together a unit test that would demonstrate the behavior, and couldn't figure out how to structure it. Eventually, I realized that I should probably retry my app with the release that was closest in time to the current code. Once I updated from `2.0.0-beta1.20574.7` to `2.0.0-beta1.21216.1` I am no longer able to replicate the behavior.

FWIW, here's the difference in the tokens between the two releases:
v20574.7
```
Command: Manual
Option: -e
Argument: kf
Argument: .\public.pgp
Option: -o
Argument: .\1725.fis
'kf' was not matched. Did you mean '-ekf', or '-r', or '-e', or '-o', or '-h'?
```

v21216.1
```
Command: Manual
Option: -ekf
Argument: .\public.pgp
Option: -o
Argument: .\1725.fis
```
